source/include/signaling_data_types.h

```diff
@@ -0,0 +1,381 @@
+ *        Refer to https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/API_DeleteSignalingChannel.html for more detail.
+ */
+typedef struct DeleteSignalingChannelRequestInfo
+{
+    SignalingChannelArn_t channelArn;
+    char * pVersion;
+    size_t versionLength;
+} DeleteSignalingChannelRequestInfo_t;
+
+/**
+ * @ingroup signaling_enum_types
+ * @brief The structure of GetSignalingChannelEndpoint request.
+ *        Refer to https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/API_GetSignalingChannelEndpoint.html for more detail.
+ */
+typedef struct ConnectWssEndpointRequestInfo
+{
+    SignalingChannelArn_t channelArn;
+    SignalingRole_t role;
+    const char * pClientId;
+    size_t clientIdLength;
+} ConnectWssEndpointRequestInfo_t;
+
+/**
+ * @ingroup signaling_enum_types
+ * @brief The structure to send message to websocket secure endpoint.
+ *        Refer to https://docs.aws.amazon.com/kinesisvideostreams-webrtc-dg/latest/devguide/kvswebrtc-websocket-apis3.html,
+ *        https://docs.aws.amazon.com/kinesisvideostreams-webrtc-dg/latest/devguide/kvswebrtc-websocket-apis4.html,
+ *        https://docs.aws.amazon.com/kinesisvideostreams-webrtc-dg/latest/devguide/kvswebrtc-websocket-apis5.html and
+ *        https://docs.aws.amazon.com/kinesisvideostreams-webrtc-dg/latest/devguide/kvswebrtc-websocket-apis6.html for more detail.
+ */
+typedef struct WssSendMessage
+{
+    SignalingTypeMessage_t messageType;
+    const char * pRecipientClientId;
+    size_t recipientClientIdLength;
+    const char * pBase64EncodedMessage;
+    size_t base64EncodedMessageLength;
+    const char * pCorrelationId;
+    size_t correlationIdLength;
+    SignalingIceServer_t * pIceServers;
+    size_t numIceServers;
+} WssSendMessage_t;
+
+/**
+ * @ingroup signaling_enum_types
+ * @brief The status response structure in receive event message from websocket secure endpoint.
+ *        Refer to https://docs.aws.amazon.com/kinesisvideostreams-webrtc-dg/latest/devguide/kvswebrtc-websocket-apis-7.html for more detail.
+ */
+typedef struct WssStatusResponse
+{
+    const char * pCorrelationId;
+    size_t correlationIdLength;
+    const char * pErrorType;
+    size_t errorTypeLength;
+    const char * pStatusCode;
+    size_t statusCodeLength;
+    const char * pDescription;
+    size_t descriptionLength;
+} WssStatusResponse_t;
+
+/**
+ * @ingroup signaling_enum_types
+ * @brief The event message structure from websocket secure endpoint.
+ *        Refer to https://docs.aws.amazon.com/kinesisvideostreams-webrtc-dg/latest/devguide/kvswebrtc-websocket-apis-7.html for more detail.
+ */
+typedef struct WssRecvMessage
+{
+    const char * pSenderClientId;
+    size_t senderClientIdLength;
+    SignalingTypeMessage_t messageType;
+    const char * pBase64EncodedPayload;
+    size_t base64EncodedPayloadLength;
+    WssStatusResponse_t statusResponse;
+    SignalingIceServer_t * pIceServers; /* The caller needs to supply memory. */

```

Comment:
can you make it consistent across all structs?

Reply:
Assuming that you are talking about memory related comment, it is already consistent across all the structs. I guess this comment is uncalled for - I have removed it.